staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -261,6 +289,28 @@ func ValidateObjectMetaAccessorUpdate(newMeta, oldMeta metav1.Object, fldPath *f
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("deletionTimestamp"), newMeta.GetDeletionTimestamp(), "field is immutable; may only be changed via deletion"))
 	}
 
+	oldInit := oldMeta.GetInitializers()
+	newInit := newMeta.GetInitializers()
+	switch {
+	case oldInit == nil && newInit != nil:
+		// Initializers may not be set on new objects
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("initializers"), nil, "field is immutable once all initialization has completed"))
```

Comment:
Although you could just call the validation function from the admission controller in this case...